internal/resources/gitrepository/resource_git_repository.go

```diff
@@ -17,26 +29,309 @@ import (
+	d.SetId(UID)
+
+	return dataSourceGitRepositoryRead(ctx, d, m)
+}
+
+func resourceGitRepositoryDelete(ctx context.Context, d *schema.ResourceData, m interface{}) (diags diag.Diagnostics) {
+	config := m.(authctx.TanzuContext)
+
+	gitRepositoryName, ok := d.Get(nameKey).(string)
+	if !ok {
+		return diag.Errorf("unable to read git repository name")
+	}
+
+	gitRepositoryNamespaceName, ok := d.Get(namespaceNameKey).(string)
+	if !ok {
+		return diag.Errorf("unable to read git repository namespace name")
+	}
+
+	scopedFullnameData := scope.ConstructScope(d, gitRepositoryName, gitRepositoryNamespaceName)
+
+	if scopedFullnameData == nil {
+		return diag.Errorf("Unable to delete Tanzu Mission Control git repository entry; Scope full name is empty")
+	}
+
+	switch scopedFullnameData.Scope {
+	case commonscope.ClusterScope:
+		if scopedFullnameData.FullnameCluster != nil {
+			err := config.TMCConnection.ClusterGitRepositoryResourceService.VmwareTanzuManageV1alpha1ClusterFluxcdGitrepositoryResourceServiceDelete(scopedFullnameData.FullnameCluster)
+			if err != nil && !clienterrors.IsNotFoundError(err) {
+				return diag.FromErr(errors.Wrapf(err, "Unable to delete Tanzu Mission Control cluster git repository entry, name : %s", gitRepositoryName))
+			}
+		}
+	case commonscope.ClusterGroupScope:
+		if scopedFullnameData.FullnameClusterGroup != nil {
+			err := config.TMCConnection.ClusterGroupGitRepositoryResourceService.VmwareTanzuManageV1alpha1ClustergroupFluxcdGitrepositoryResourceServiceDelete(scopedFullnameData.FullnameClusterGroup)
+			if err != nil && !clienterrors.IsNotFoundError(err) {
+				return diag.FromErr(errors.Wrapf(err, "Unable to delete Tanzu Mission Control cluster group git repository entry, name : %s", gitRepositoryName))
+			}
+		}
+	case commonscope.UnknownScope:
+		return diag.Errorf("no valid scope type block found: minimum one valid scope type block is required among: %v. Please check the schema.", strings.Join(scope.ScopesAllowed[:], `, `))
+	}
+
+	// d.SetId("") is automatically called assuming delete returns no errors, but
+	// it is added here for explicitness.
+	d.SetId("")
+
+	return diags
+}
+
+func resourceGitRepositoryInPlaceUpdate(ctx context.Context, d *schema.ResourceData, m interface{}) (diags diag.Diagnostics) {
+	config := m.(authctx.TanzuContext)
+
+	gitRepositoryName, ok := d.Get(nameKey).(string)
+	if !ok {
+		return diag.Errorf("unable to read git repository name")
+	}
+
+	gitRepositoryNamespaceName, ok := d.Get(namespaceNameKey).(string)
+	if !ok {
+		return diag.Errorf("unable to read git repository namespace name")
+	}
+
+	scopedFullnameData := scope.ConstructScope(d, gitRepositoryName, gitRepositoryNamespaceName)
+
+	if scopedFullnameData == nil {
+		return diag.Errorf("Unable to update Tanzu Mission Control git repository entry; Scope full name is empty")
+	}
+
+	// nolint: dogsled
+	_, meta, atomicSpec, _, _, err := retrieveGitRepositoryUIDMetaAndSpecFromServer(config, scopedFullnameData, d)
```

Comment:
reduce the number of parameters being returned here. 6 return parameters is not acceptable

Reply:
Updated.